src/dynamics/periodic_pts.lean

```diff
@@ -208,6 +209,16 @@ begin
   { exact is_periodic_pt_zero f x }
 end
 
+lemma iterate_eq_mod_minimal_period : f^[n] x = (f^[n % minimal_period f x] x) :=
+begin
+  conv_lhs { rw ← nat.mod_add_div n (minimal_period f x) },
+  rw [iterate_add, mul_comm, iterate_mul],
+  simp only [comp_app],
+  congr,
+  rw [← is_fixed_pt, ← is_periodic_pt],
+  exact is_periodic_pt.iterate (is_periodic_pt_minimal_period _ _) _,
```

Comment:
```suggestion
  conv_lhs { rw ← nat.mod_add_div' n (minimal_period f x) },
  rw [iterate_add, iterate_mul, comp_app],
  congr,
  exact is_periodic_pt.iterate (is_periodic_pt_minimal_period _ _) _,
```
Shorter proof.

Reply:
This lemma should probably be about natural numbers being equal iff they divide the same numbers.

If Yoneda's lemma is in the category theory part of the library, we might prove this using the partial order under divisibility as a category on `nat`. (-;